apps/snitch_core/test/domain/order/transitions_test.exs

```diff
@@ -141,4 +149,49 @@ defmodule Snitch.Domain.Order.TransitionsTest do
       assert {:ok, %{packages: []}} = Repo.transaction(result.multi)
     end
   end
+
+  describe "associate_package" do
+    setup :shipping_categories
+    setup :zones
+    setup :shipping_methods_embedded
+
+    @tag shipping_category_count: 1,
+         shipping_method_count: 1,
+         state_zone_count: 1
+    setup context do
+      expect(Snitch.Tools.DefaultsMock, :fetch, 3, fn :currency -> {:ok, :USD} end)
+      order = insert(:order, user: build(:user))
+      %{shipping_methods: [sm]} = context
+
+      packages =
+        insert_list(
+          1,
+          :package,
+          order_id: order.id,
+          origin: build(:stock_location),
+          shipping_methods: [sm],
+          shipping_category: build(:shipping_category)
+        )
+
+      [order: order, packages: packages]
+    end
+
+    test "with packages", %{order: order, packages: packages} do
+      package = List.first(packages)
+      selected_shipping_method = List.first(package.shipping_methods)
+
+      shipping_methods = [
+        %{package_id: package.id, shipping_method_id: selected_shipping_method.id}
+      ]
+
+      result =
+        order
+        |> Context.new(state: %{selected_shipping_methods: shipping_methods})
+        |> Transitions.save_packages_methods()
+
+      assert result.valid?
+      assert [packages: {:run, _}] = Multi.to_list(result.multi)
+      assert {:ok, %{packages: _}} = Repo.transaction(result.multi)
```

Comment:
1. Added

2. I think we can never arrive at this scenario, order without packages, at delivery.

Reply:
Unfortunately, (2) is possible. Say an `order` is in `address` state and the user goes to the cart and empties it. According to [the epic I'm working on], we'll still keep the order is `address` state, and this order has no packages.
There are some other scenarios where this can happen as well.